New Entry Cloud IaaS (#932)
Codecademy/docs: content/cloud-computing/concepts/iaas/iaas.md

---
Title: 'IaaS'

> Description: 'Infrastructure as a service (IaaS) is a type of cloud computing service offering remote computer, storage and networking resources that are available on demand'

Reviewer: ```suggestion
Description: 'Infrastructure as a service (IaaS) cloud computing model that offers remote computer, storage, and networking resources that are available on demand'
```

Subjects:
  - 'Computer Science'
  - 'Networking'
Tags:
  - 'Networking'
  - 'Cloud'
  - 'Infrastructure'
CatalogContent:

>   - 'learn/foundations-of-cloud-computing'

Reviewer 2: ```suggestion
  - 'foundations-of-cloud-computing'
```

Thinking this is the correct format, also if you see this entry: https://www.codecademy.com/resources/docs/general/routing 

I think the first catalog content should be (strangely) `'cybersecurity-intro-to-it'`

  - 'paths/back-end-engineer-career-path'
---

> **Infrastructure as a service (IaaS)** is a type of cloud computing service offering remote computer, storage and networking resources that are available on demand. These services are typically offered on a pay-as-you-go basis. The user is given control of virtual machines that are guests on a cloud server running a hypervisor such as Oracle VirtualBox or VMware ESXi. They are allowed to run arbitrary software on these virtual machines, including operating systems. They are also provided with an API to control the low-level details of the underlying network infrastructure: things such as backups, partitioning, security, physical resources and so on. While the user does not manage or control the underlying cloud infrastructure, they have granular control over operating systems, storage, deployed applications, as well as some networking components such as host firewalls.

Reviewer: ```suggestion
**Infrastructure as a service (IaaS)** is a cloud computing model that offers remote computer, storage, and networking resources that are available on demand. These services are typically offered on a pay-as-you-go basis. The user is given control of virtual machines that are guests on a cloud server running a hypervisor such as Oracle VirtualBox or VMware ESXi. They are allowed to run arbitrary software on these virtual machines, including operating systems.

IaaS is also provided with an [API](https://www.codecademy.com/resources/docs/general/api) to control the low-level details of the underlying network infrastructure: things such as backups, partitioning, security, physical resources, and so on. While the user does not manage or control the underlying cloud infrastructure, they have granular control over operating systems, storage, deployed applications, as well as some networking components such as host firewalls.
```

Reviewer: Given that we're working on a separate "Types of Cloud" entry in another PR, I think we should avoid this phrasing for these entries to mitigate any possible confusion.

---
Title: 'IaaS'
Description: 'Infrastructure as a service (IaaS) is a cloud computing model that offers remote computer, storage, and networking resources that are available on demand.'
Subjects:
  - 'Computer Science'
  - 'Networking'
Tags:
  - 'Networking'
  - 'Cloud'
  - 'Infrastructure'
CatalogContent:
  - 'foundations-of-cloud-computing'
  - 'paths/back-end-engineer-career-path'
---

> **Infrastructure as a service (IaaS)** is a cloud computing model that offers remote computer, storage, and networking resources that are available on demand. These services are typically offered on a pay-as-you-go basis. The user is given control of virtual machines that are guests on a cloud server running a hypervisor such as Oracle VirtualBox or VMware ESXi. They are allowed to run arbitrary software on these virtual machines, including operating systems.

Reviewer 3: Several terms here are not adequately explained. Keep in mind readers might not be familiar with terms such as pay as you go, virtual machine or hypervisor.

Reviewer 3: Recommend attempting to simplify

Reviewer: I'd also recommend, for a future PR, we make new General concept entries for hypervisor and virtual machine.

> IaaS is also provided with an [API](https://www.codecademy.com/resources/docs/general/api) to control the low-level details of the underlying network infrastructure: things such as backups, partitioning, security, physical resources, and so on. While the user does not manage or control the underlying cloud infrastructure, they have granular control over operating systems, storage, deployed applications, as well as some networking components such as host firewalls.

Reviewer 3: IaaS provides an API maybe instead of is also provided by. Also this first sentence is run on. 

IaaS provides an [API](https://www.codecademy.com/resources/docs/general/api) to control the low-level details of the underlying infrastructure. While they do not have direct access to the infrastructure, users can configure:

Reviewer 3: Followed by a list of the things you list in the paragraph.

Reviewer: ```suggestion
IaaS provides an [API](https://www.codecademy.com/resources/docs/general/api) to control the low-level details of the underlying network infrastructure (e.g., backups, partitioning, security, and physical resources). While they do not have direct access to the infrastructure, users can configure:

- Operating systems
- Storage
- Deployed applications
- Networking components (e.g., host firewalls)
```

> The user is responsible for maintaining the virtual machines, including keeping operating systems and applications patched and up-to-date, and pays the cloud service only for the number of resources allocated and consumed.

Reviewer 3: Break up into a list that the user is responsible for

Reviewer: ```suggestion
The user's responsibilities include the following:

- Maintaining the virtual machines.
  - This includes keeping operating systems and applications patched and up-to-date.
- Paying the cloud service only for the number of resources allocated and consumed.
```
